app/code/Magento/InventorySourceSelectionApi/Model/GetAvailableSourceItemsBySkusAndSortedSource.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySourceSelectionApi\Model;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\SourceItemRepositoryInterface;
+
+/**
+ * Retrieve source items for a defined set of skus and sorted source codes
+ *
+ * Useful for determining availability and source selection
+ */
+class GetAvailableSourceItemsBySkusAndSortedSource
```

Comment:
There is no defined meaning for `Available Source Item` in the scope of Inventory. Should we change `available` to something in the lines of `InStock`?